src/Shipping/ZoneMethodsParser.php

```diff
@@ -123,7 +123,9 @@ protected function get_flat_rate_method_rate( object $method ): ?float {
 		$rate = null;
 
 		$flat_cost = 0;
-		$cost      = $method->get_option( 'cost' );
+		$locale    = localeconv();
+		$decimals  = [ wc_get_price_decimal_separator(), $locale['decimal_point'], $locale['mon_decimal_point'], ',' ];
+		$cost      = str_replace( $decimals, '.', (string) $method->get_option( 'cost' ) );
```

Comment:
This looks like a great candidate for a helper function, both to clarify what is being done here as well as making it reusable. It also seems like WC already has [wc_format_decimal](https://woocommerce.github.io/code-reference/files/woocommerce-includes-wc-formatting-functions.html#function_wc_format_decimal). Should we add that to the WC proxy class, so we can simplify testing as a bonus?

Reply:
I tried using the `wc_format_decimal` function, but there is an issue. If your decimal separator is `.` and you use `wc_format_decimal('2,4', '')`, it returns `24.00`, which is incorrect. The issue is that `wc_format_decimal` doesn't specifically replace the comma; it just uses the decimal settings stored in WooCommerce and the local environment:

https://github.com/woocommerce/woocommerce/blob/91afd71b96e55338affb9284d22df0c38297ee09/plugins/woocommerce/includes/wc-formatting-functions.php#L292

On the other hand, the Shipping Flat Rate function explicitly handles the comma:

https://github.com/woocommerce/woocommerce/blob/91afd71b96e55338affb9284d22df0c38297ee09/plugins/woocommerce/includes/shipping/flat-rate/class-wc-shipping-flat-rate.php#L86

Reply:
Thanks for clarifying that I failed to see the difference of the extra `,` being added to the array.

My question though, is that the right way to handle it? Because if we enter a shipping rate of `2,399.99`, then it will be evaluated as `2.399.99`, which I don't think is correct either, and won't pass an `is_numeric` test. If I have a store setup with `.` as the decimal separator and ',' as the thousand separator, then I don't expect it to be replacing `,` for decimals. So in that case I'd see 2,4 being replaced with 24 as the "more correct" way of handling it.

Reply:
Thanks for the discussion. It's a tricky situation because the issue arises from the flexibility users have when setting prices in the modal. It allows any characters as decimal or thousand separators.

The solution I proposed aligns best with how the current WC Shipping Flat Rate handles these cases, although it was not ideal. However, I’ve adjusted the function to strip out any thousand separators and convert the string into a valid decimal number. For example, 1.234,01 will be converted to 1234.01. The thousand separator is more of a formatting issue and doesn’t affect syncing rates or operating with the numbers itself. If we want to format the number, we can use the string returned from the function and apply any desired decimal or thousand separators.

Reply:
Thanks for the clarification, it is indeed tricky.

Although I don't totally agree with the validation of the modal as if I leave it to the default with no thousand separator then I'm only allowed to use a `.` and a comma is not allowed.
![image](https://github.com/user-attachments/assets/26d5883e-9be7-4a90-b765-6230a6671dfb)

If I do have both symbols included then it accepts any combination as long as they are either numbers or a thousand separator, and a single decimal separator. So entering `1,,23.45,,99` is totally acceptable even though it won't make sense. In your case that will be converted to 12345.99

That's fine for now as generally we will expect sane values being entered, but I think in the long run it's the modal that should do a bit better validation there.